cli/filter_data_through_cmd.go

```diff
@@ -0,0 +1,93 @@
+package cli
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func filterDataThroughCmd(data []byte, filter string, subject string) ([]byte, error) {
+	parts, err := parseCommandLine(filter)
+	if err != nil {
+		return nil, fmt.Errorf("the filter command line could not be parsed: %w", err)
+	}
+	cmd := parts[0]
+	args := parts[1:]
+	// maybe we want to replace some strings to give the filter something to reason about?
+	for idx, arg := range args {
+		args[idx] = strings.ReplaceAll(arg, "{{subject}}", subject)
+	}
+	runner := exec.Command(cmd, args...)
+	// pass the message as string to stdin
+	runner.Stdin = bytes.NewReader(data)
+	// maybe we want to do something on error?
+	return runner.CombinedOutput()
+}
+
+func parseCommandLine(command string) ([]string, error) {
```

Comment:
maybe https://github.com/google/shlex can be used, despite being archived its still useful think it does what you're doing here

Reply:
I was not sure about adding dependencies. But you are right, it will be much better to use this as it has a test and is doing exactly what it is used for here.